File: domain/src/testFixtures/java/com/study/platform/support/fake/FakeNotificationPublisher.java
package com.study.platform.support.fake;

import com.study.platform.domain.notification.model.NotificationPublisher;
import com.study.platform.domain.notification.model.NotificationType;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class FakeNotificationPublisher implements NotificationPublisher {

    public record SentNotification(UUID receiverId, NotificationType type, String message) {}

    private final List<SentNotification> sent = new ArrayList<>();

    @Override
    public void send(Long outboxEventId, UUID receiverId, NotificationType type, String message, UUID targetId) {
        sent.add(new SentNotification(receiverId, type, message));
    }

    public List<SentNotification> getSent() {
        return sent;
    }

    public boolean hasSentTo(UUID receiverId) {
        return sent.stream().anyMatch(n -> n.receiverId().equals(receiverId));
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeNotificationPublisher.kt
package com.study.platform.support.fake

import com.study.platform.domain.notification.model.NotificationPublisher
import com.study.platform.domain.notification.model.NotificationType
import java.util.UUID

class FakeNotificationPublisher : NotificationPublisher {

    data class SentNotification(val receiverId: UUID, val type: NotificationType, val message: String)

    private val sent: MutableList<SentNotification> = ArrayList()

    override fun send(outboxEventId: Long, receiverId: UUID, type: NotificationType, message: String, targetId: UUID?) {
        sent.add(SentNotification(receiverId, type, message))
    }

    fun getSent(): List<SentNotification> = sent

    fun hasSentTo(receiverId: UUID): Boolean =
        sent.any { n -> n.receiverId == receiverId }
}
